server/platform/services/cache/cache.go

```diff
@@ -40,9 +40,6 @@ type Cache interface {
 	// Keys returns a slice of the keys in the cache.
 	Keys() ([]string, error)
 
-	// Len returns the number of items in the cache.
-	Len() (int, error)
```

Comment:
Would removing this make it harder to track cache size as a performance metric?

Reply:
Yeah, but the thing is we cannot track the cache size in Redis as a prom counter. Because it's just too expensive. It's one of the tradeoffs we have to make.

Additionally, max cache size cannot be implemented at a singular cache level at Redis. But more at a global level, where if the total size of all caches combined exceed maxMemory, then Redis triggers eviction based on LRU.

So in that aspect, I don't think tracking individual cache size is going to be of much benefit (atleast wrt Redis). We have to look a bit more into invalidations and how we can optimize that from the code.